tests/indexes/tests.py

```diff
@@ -178,3 +232,117 @@ def test_no_index_for_foreignkey(self):
             if field_created:
                 with connection.schema_editor() as editor:
                     editor.remove_field(ArticleTranslation, new_field)
+
+
+@override_settings(USE_TZ=True)
+@skipUnlessDBFeature('supports_partial_indexes')
+class PartialIndexTests(TestCase):
+    # In the tests, the schema editor is asked to actually create the index
+    # to ensure that it works.
+
+    def test_partial_index(self):
+        with connection.schema_editor() as editor:
+            index = Index(
+                name='recent_article_idx',
+                fields=['pub_date'],
+                condition=Q(
+                    pub_date__gt=datetime.datetime(year=2015, month=1, day=1, tzinfo=timezone.get_current_timezone()),
```

Comment:
Do we need the `tzinfo` bit for the test? I'm worried relying on `get_current_timezone` could make the test flaky.

Reply:
The reason for the `tzinfo` bit was the type casting which makes the function mutable, and PostgreSQL raises an error. `USE_TZ` is `False` by default; without `tzinfo`, the test fails with `django.db.utils.ProgrammingError: functions in index predicate must be marked IMMUTABLE`. I believed I touched this problem about timezones in the documentation, but it may need improvement or clarification.

Reply:
Maybe include a comment here?

Reply:
Yes :+1: I'll also try to write a sentence or two about `DateField` under the postgresql note. I noticed that comparison between dates (and not date+time) won't be converted to a mutable function.